### PR TITLE
enhancement: added shimmer effect

### LINKED ISSUE #875: Enhancement: Add shimmer effect to the still loading part of the extension requests card
### Issue Description
Currently in the extension requests card when it is loading we see a slight delay in some of the elements being rendered, during that time when those element are being rendered , there is nothing in the UI to show that its being rendered . There is just a blank. This can be seen in the screenshot that i have shared. The data that has already being loaded by the card is inaccessible to the user because of the loading spinner. Hence the loading spinner should be removed and there should be a shimmer in the place of the data that is still being loaded.
Screenshot from 2024-10-06 20-20-13
There should be a shimmer in the place of blanks like this 
Screenshot from 2024-10-06 20-23-08

### Screenshots
[Screencast from 06-10-24 09:58:48 AM IST.webm](https://github.com/user-attachments/assets/67f93a2e-7f0f-4926-8a31-67d56bf2fd49)

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [x] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by rishirishhh**
Date: 12/10/2024

Developer Name: Kavya Rishi 

---

## Issue Ticket Number 

- Closes #875 
## Description
Currently in the extension requests card when it is loading we see a slight delay in some of the elements being rendered, during that time when those element are being rendered , there is nothing in the UI to show that its being rendered . There is just a blank. This can be seen in the screenshot that i have shared. The data that has already being loaded by the card is inaccessible to the user because of the loading spinner. Hence the loading spinner should be removed and there should be a shimmer in the place of the data that is still being loaded.

### Documentation Updated?

- [ ] Yes
- [X] No

### Under Feature Flag

- [X] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [X] No

### Breaking Changes

- [ ] Yes
- [X] No

### Development Tested?

- [X] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

[Screencast from 16-10-24 12:37:48 PM IST.webm](https://github.com/user-attachments/assets/f9d2d63d-1e62-4848-bcc3-9d2956f8f318)

</details>

## Test Coverage
Screenshot from 2024-10-12 22-25-05

### CODE REVIEW

File: extension-requests/script.js [hunk 1]
```diff
@@ -1198,10 +1258,22 @@ async function createExtensionCard(data) {
     userFirstName = userFirstName ?? '';
     statusSiteLink.href = `${STATUS_BASE_URL}/tasks/${data.taskId}`;
     statusSiteLink.innerText = taskData.title;
+    if (dev) {
+      statusSiteLink.classList.remove('skeleton-link');
```
Comment (joyguptaa) [inline extension-requests/script.js:1262]: nit : `assigneeNameElement.classList.toggle('skeleton-text', !dev);`

Comment (joyguptaa) [reply extension-requests/script.js:1262]: ref : https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle

Comment (joyguptaa) [reply extension-requests/script.js:1262]: same for line 1266, 1271, 1275

File: extension-requests/style.css [hunk 2]
```diff
@@ -166,6 +166,40 @@
   background-color: var(--green500);
 }
 
+.skeleton {
+  width: 1.5rem; /* Adjust the width */
```
Comment (joyguptaa) [inline extension-requests/style.css:170]: remove the comments from this file

**Comment by joyguptaa** [comment]
The test Coverage Screenshot is not visible. Please Check

**Comment by rishirishhh** [comment]
Screenshot from 2024-10-12 22-25-05
Test Coverage

**Comment by shubhdevelop** [comment]
> Screenshot from 2024-10-12 22-25-05 Test Coverage

add it to the Pr description it's still not visible there

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by shubhdevelop — DISMISSED** [review]

**Comment by joyguptaa** [comment]
> Screenshot from 2024-10-12 22-25-05 Test Coverage

Please add coverage screenshot for extension-requests/script.js file as it is not visible in the attached image.

**Comment by rishirishhh** [comment]
> > Screenshot from 2024-10-12 22-25-05 Test Coverage
> 
> Please add coverage screenshot for extension-requests/script.js file as it is not visible in the attached image.

Screenshot from 2024-10-16 00-10-41
Here is the image @joyguptaa

**Review by joyguptaa — DISMISSED** [review]

**Review by iamitprakash — CHANGES_REQUESTED** [review]
make design uniform

**Comment by shubhdevelop** [comment]
update the video in PR, reflecting latest changes

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**